src/core/mathsEvaluator.js

```diff
@@ -54,16 +53,60 @@ const defaultDecimalConfig = {
 };
 
 /**
- * List of config entries the Decimal constructor accepts
- * @type {string[]}
+ * Checks if an object is empty.
+ * @param {*} obj - The object to check.
+ * @returns {boolean} - True if the object is empty, false otherwise.
  */
-const decimalConfigEntries = ['precision', 'rounding', 'toExpNeg', 'toExpPos', 'maxE', 'minE', 'modulo', 'crypto'];
+const isEmpty = obj => [Object, Array].includes((obj || {}).constructor) && !Object.entries(obj || {}).length;
 
 /**
- * List of config entries the Parser constructor accepts
- * @type {string[]}
+ * Checks if an object is a plain object.
+ * @param {*} obj - The object to check.
+ * @returns {boolean} - True if the object is a plain object, false otherwise.
  */
-const parserConfigEntries = ['operators'];
+const isPlainObject = obj => [Object].includes((obj || {}).constructor) && Object.entries(obj || {}).length;
```

Comment:
Not sure to understand your point. Do you mean to use a single implementation or factorize it for isEmpty and isPlainObject? Actually, they slightly differ. I'm not sure it will be of huge benefit to optimize it.